src/todos/todos.change-stream.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common'
+import { InjectModel } from '@nestjs/mongoose'
+import { Model } from 'mongoose'
+import type { ChangeStream } from 'mongodb'
+
+import { TODO_MODEL, USER_MODEL } from '../common/constants/database.constants'
+import type { Todo } from './schemas/todos.schema'
+import type { User } from '../user/schemas/user.schema'
+
+// TODO: - Refactor User and Todo change stream services using abstract class to reuse common logic;
+
+@Injectable()
+export class TodosChangeStreamService implements OnModuleInit, OnModuleDestroy {
+  private changeStream: ChangeStream
+  private resumeToken: unknown
+
+  constructor(
+    @InjectModel(TODO_MODEL) private todoModel: Model<Todo>,
+    @InjectModel(USER_MODEL) private userModel: Model<User>,
+  ) {}
+
+  async onModuleInit() {
+    this.openChangeStream()
+  }
+
+  async onModuleDestroy() {
+    this.closeChangeStream()
+  }
+
+  private async openChangeStream() {
```

Comment:
Let's rename so that the name will better reflect method nature.

Reply:
What about `private addChangeStreamListener() {...}`?